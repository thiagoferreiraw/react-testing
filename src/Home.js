import React from 'react';
import PropTypes from "prop-types"

const Home = ({ username, tasks }) => (
  <div>
    <p>Welcome, {username}!</p>
    <ul>
      {tasks && tasks.map((t, idx) => <li key={idx}>{t}</li>)}
    </ul>
  </div>
)

Home.propTypes = {
  username: PropTypes.string,
  tasks: PropTypes.array
}

export default Home
