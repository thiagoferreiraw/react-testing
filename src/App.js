import React, { Component } from 'react';
import PropTypes from "prop-types"
import Home from "./Home"

const fakeDatabase = {
  "Heisenberg": [
    "8am - Chemistry classes at school",
    "12:30pm - Meet Jesse for lunch",
    "15pm - Meet Gus at the Pollos Hermanos",
    "20pm - Dinner by the pool with Hank and Marie"
  ],
  "GusFring": [
    "8am - Meeting with the DEA",
    "15pm - Meet Walter at the Pollos Hermanos",
  ]
}

class App extends Component {
  static propTypes = {
    isLoggedIn: PropTypes.bool,
    username: PropTypes.string
  }

  render() {
    const { isLoggedIn, username } = this.props

    return (
      <div className="App">
        {
          isLoggedIn
            ? <Home username={username} tasks={fakeDatabase[username]} />
            : <p>Hello, visitor. Sign in to continue.</p>
        }
      </div>
    );
  }
}

export default App;
