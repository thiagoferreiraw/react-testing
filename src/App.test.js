import React from 'react';
import App from './App';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json"
import Home from './Home';

it("should render the homepage for the non logged in user", () => {
  const props = {
    isLoggedIn: false,
    username: null
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()

  expect(app.find(Home)).toHaveLength(0)
  expect(app.find('p').at(0).text()).toEqual("Hello, visitor. Sign in to continue.")
})

it("should render the Homepage for the logged in user (Heisenberg)", () => {
  const props = {
    isLoggedIn: true,
    username: "Heisenberg"
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()

  expect(app.find(Home)).toHaveLength(1)
  expect(app.find('p').at(0).text()).toEqual("Welcome, Heisenberg!")

  expect(app.find('li')).toHaveLength(4)
  expect(app.find('li').at(0).text()).toEqual("8am - Chemistry classes at school")
  expect(app.find('li').at(1).text()).toEqual("12:30pm - Meet Jesse for lunch")
  expect(app.find('li').at(2).text()).toEqual("15pm - Meet Gus at the Pollos Hermanos")
  expect(app.find('li').at(3).text()).toEqual("20pm - Dinner by the pool with Hank and Marie")
})

it("should render the Homepage for the logged in user without tasks", () => {
  const props = {
    isLoggedIn: true,
    username: "Jesse"
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()

  expect(app.find(Home)).toHaveLength(1)
  expect(app.find('p').at(0).text()).toEqual("Welcome, Jesse!")

  expect(app.find('li')).toHaveLength(0)
})