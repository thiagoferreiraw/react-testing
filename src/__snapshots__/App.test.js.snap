// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Homepage for the logged in user (Heisenberg) 1`] = `
<App
  isLoggedIn={true}
  username="Heisenberg"
>
  <div
    className="App"
  >
    <Home
      tasks={
        Array [
          "8am - Chemistry classes at school",
          "12:30pm - Meet Jesse for lunch",
          "15pm - Meet Gus at the Pollos Hermanos",
          "20pm - Dinner by the pool with Hank and Marie",
        ]
      }
      username="Heisenberg"
    >
      <div>
        <p>
          Welcome, 
          Heisenberg
          !
        </p>
        <ul>
          <li
            key="0"
          >
            8am - Chemistry classes at school
          </li>
          <li
            key="1"
          >
            12:30pm - Meet Jesse for lunch
          </li>
          <li
            key="2"
          >
            15pm - Meet Gus at the Pollos Hermanos
          </li>
          <li
            key="3"
          >
            20pm - Dinner by the pool with Hank and Marie
          </li>
        </ul>
      </div>
    </Home>
  </div>
</App>
`;

exports[`should render the Homepage for the logged in user without tasks 1`] = `
<App
  isLoggedIn={true}
  username="Jesse"
>
  <div
    className="App"
  >
    <Home
      username="Jesse"
    >
      <div>
        <p>
          Welcome, 
          Jesse
          !
        </p>
        <ul />
      </div>
    </Home>
  </div>
</App>
`;

exports[`should render the homepage for the non logged in user 1`] = `
<App
  isLoggedIn={false}
  username={null}
>
  <div
    className="App"
  >
    <p>
      Hello, visitor. Sign in to continue.
    </p>
  </div>
</App>
`;
